### ISSUE TITLE
UserWarning triggered by scheduler.step(0) inside the step function in SequentialLR class

### ISSUE THREAD

**Issue by rogaits** [pytorch/pytorch#130018]
### 🐛 Describe the bug

The warning message only appears the first time the code snippet below is run. 
```
~/torch/optim/lr_scheduler.py:156: UserWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
  warnings.warn(EPOCH_DEPRECATION_WARNING, UserWarning)
```

code snippet
``` python
import torch
import matplotlib.pyplot as plt

model = torch.nn.Linear(5,5)
nepochs = 3
warm_up_epochs = 1
optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9)
scheduler2 = torch.optim.lr_scheduler.CosineAnnealingLR(optimizer, T_max=nepochs-warm_up_epochs, eta_min=1e-5)
scheduler1 = torch.optim.lr_scheduler.LinearLR(optimizer, start_factor=0.3,end_factor=1,total_iters=warm_up_epochs)
scheduler = torch.optim.lr_scheduler.SequentialLR(optimizer, schedulers=[scheduler1, scheduler2], milestones=[warm_up_epochs])
x = torch.randn(8,2,5)
y = torch.randn(x.size())
loss_fn = torch.nn.MSELoss()

for epoch in range(1,nepochs+1):
    model.train(True)
    y_pred = model(x)
    loss = loss_fn(y_pred,y)
    optimizer.zero_grad()
    loss.backward()
    optimizer.step()

    scheduler.step()

    print(epoch,optimizer.param_groups[0]["lr"])

```

I suspect the issue might be here https://github.com/pytorch/pytorch/blob/8f70bf7a943799b5cd870952d39f36361de4b87f/torch/optim/lr_scheduler.py#L919
as it was also mentioned in a forum post
https://discuss.pytorch.org/t/why-theres-a-line-scheduler-step-0-in-the-code-of-class-sequentiallr/194794

### Versions

```
Versions of relevant libraries:
[pip3] numpy==1.26.3
[pip3] torch==2.2.0+cu118
[pip3] torchaudio==2.2.0+cu118
[pip3] torchvision==0.17.0+cu118
[pip3] triton==2.2.0
```

**Comment by ZhouSky** [comment]
i'm wondering why not temporarily filter this warning in SequentialLR.step

**Comment by janeyx99** [comment]
@ZhouSky We don't have a dedicated expert on LRScheduler, but I would review the temporary filter PR.

**Comment by ZhouSky** [comment]
I haven't read too much discussion about lr_scheduler. For this small problem, here is my solution. 
```
import warnings

from torch.optim.lr_scheduler import SequentialLR as _SequentialLR 
from torch.optim.lr_scheduler import EPOCH_DEPRECATION_WARNING

class SequentialLR(_SequentialLR):
    def step(self, epoch=...):
        with warnings.catch_warnings():  # not thread-safe
            warnings.filterwarnings('ignore', message=EPOCH_DEPRECATION_WARNING[:10], category=UserWarning)
            super().step()
```
Maybe we can wrap `scheduler.step(0)` in `SequentialLR.step` in the same way like below? I'm not sure if it is appropriate, or need design avoid using `scheduler.step(0)`.
```
class SequentialLR(LRScheduler):
    ...
    def step(self):
        ...
            with warnings.catch_warnings():
                warnings.filterwarnings('ignore', message=EPOCH_DEPRECATION_WARNING[:10], category=UserWarning)
                scheduler.step(0)
        ...
    ...
```

**Comment by janeyx99** [comment]
If you open a PR with a test validating this behavior, I'll review.
